### PR TITLE
Fixed infinite loop bug

### PR DESCRIPTION
**Pull request by Quaker762**
-Fixed up a few opcodes.
-Tetris no longer gets caught in some weird loop.
-Added a hack to MMU so that writes to 0xFF80 are ignored.

### CODE REVIEW

**Comment by wornwinter** [comment]
I'll review this ASAP and get back to you. Could you give me a screenshot of what happens now?

File: src/DMGCPU.cpp [hunk 1]
```diff
@@ -633,17 +630,17 @@ void c_DMGCPU::OPCode0x0D()
...
-    if((Registers.BC.lo & 0xF) == 0xF)
-        SET_FLAG_BIT(HC_BIT);
-    else
+    if(Registers.BC.lo & 0x0F)
         UNSET_FLAG_BIT(HC_BIT);
+    else
+        SET_FLAG_BIT(HC_BIT);
 
-    Registers.BC.lo--;
+    Registers.BC.lo -= 1;
```
Comment (wornwinter) [inline src/DMGCPU.cpp]: You can just leave these increments or decrements as ++ or --. They're the same as -=1 or +=1

**Comment by Quaker762** [comment]
it_works

It hangs here. According to that site I found last night, it's because the write to 0xFF80 fails and there's some weird VBLANK think to fix. Anyway, it's in a much more predictable state now!

File: src/DMGCPU.cpp (hunk 1 above)
Comment (Quaker762) [inline src/DMGCPU.cpp]: Ah yep sorry. I was just cutting and pasting this morning. I'm not quite sure why I changed that... 0_o

**Comment by wornwinter** [comment]
It'll just be an unimplemented GPU register then I'd imagine, or something along those lines. Should be trivial enough.

**Comment by wornwinter** [comment]
I'm happy with this. Will merge when I'm on the computer.

**Pull request merged**